### PR TITLE
feat: support DeepSeek-V3.2 Python MTP ACLGraph.

### PR DESCRIPTION
**Pull request by yt-zeng**
## Description

This PR adds DeepSeek-V3.2 Python MTP and NPU ACLGraph support.

 The main changes include:

  - Add the DeepSeek-V3.2 MTP Python model and registry entry.
  - Share target-model embeddings and LM head with the Python MTP draft model.
  - Support multi-token MTP metadata expansion for repair/current rows.

These changes enable Python-based DeepSeek-V3.2 MTP execution with NPU ACLGraph while avoiding metadata shape mismatches during multi-request and multi-token verification.

## Change Type

- [ ] Bug fix
- [x] New feature
- [ ] Performance improvement
- [ ] Refactor
- [ ] Documentation
- [ ] Test
- [ ] Build or CI

## Performance Comparision (from server log)
DeepSeek-V3.2-w8a8-MTP, ~2K input + 1K output, 10 requests
| Metric | C++ | Python ACLGraph | Delta |
|:---|---:|---:|:---|
| TTFT avg (ms) | 823.30 | 814.30 | Python 1.1% faster |
| TPOT avg (ms/token) | 30.03 | 37.69 | C++ 20.3% faster |
| Latency avg (s) | 31.546 | 39.379 | C++ 19.9% faster |
| Decode throughput (tokens/s) | 33.36 | 26.56 | C++ 25.6% faster |

### CODE REVIEW

File: xllm/python/models/deepseek_v32_mtp.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+            if self._state_dict.has(alias):
+                return self._state_dict.get_tensor(alias)
+        raise KeyError(name)
+
+
+class DeepseekV32MtpForCausalLM(DeepseekV3ForCausalLM):
+    """DeepSeek-V3.2 MTP calculator; scheduling stays in the C++ worker."""
+
+    def __init__(self, config: dict) -> None:
+        super().__init__(config)
```
Comment (zhang-minchao) [inline xllm/python/models/deepseek_v32_mtp.py]: [P2] 请避免为 MTP 草稿模型构造完整的目标模型。这里调用父类构造函数会在 NPU 上分配一个普通的 `DeepseekV3Model`、对应的 embedding 和词表大小的 LM Head；下一行又会在旧模型主体尚未释放时构造 MTP body，而未使用的 LM Head 会一直保留到草稿模型加载完成、`PyCausalLM::share_weights_from()` 执行后。仅 BF16 LM Head 就约占用 1.85 GB / TP，临时存在的完整 MoE body 还会进一步提高启动峰值内存。原 C++ MTP 路径不会产生这些分配：它只构造 MTP 独有模块，将 `lm_head_` 保持为空，不加载草稿 embedding，加载完成后再绑定目标模型的 `NpuLmHead`/`NpuWordEmbedding` 句柄。请将公共配置初始化与目标模块分配拆开，这里只初始化 `cfg`/`dtype`/`device` 以及 MTP 独有 body；最好同时延迟创建 `lm_head` 和 `embed_tokens`，加载时跳过这两项，直到现有 C++ 权重共享步骤安装目标模块。

**Review by zhang-minchao — COMMENTED** [review]
发现两个合入前需要解决的问题。

**Review by zhang-minchao — APPROVED** [review]

**Review by yingxudeng — APPROVED** [review]
